agent/agent.go

```diff
@@ -164,20 +169,23 @@ type agent struct {
 	// Defaults to statusQueryReplyTimeout if unspecified
 	statusQueryReplyTimeout time.Duration
 
-	// memberIndex is used to select the next member to collect timeline data
-	// from.
-	// Type uint16 so limited to 0 - 65535. Assuming clusters will not have
-	// more than that many members.
-	memberIndex uint16
+	// lastSeen keeps track of the last seen timestamp from a cluster member.
+	// The last seen timestamp can be queried by a member and be used to
+	// filter out events that have already been recorded by this member.
+	lastSeen map[string]time.Time
 
 	// Config is the agent configuration.
 	Config
 
 	// SerfClient provides access to the serf agent.
 	SerfClient SerfClient
 
-	// Timeline keeps track of status change events.
+	// Timeline keeps track of all timeline events in the cluster. This timeline
+	// is only used on members that have the role 'master'.
 	Timeline history.Timeline
+
+	// LocalTimeline keeps track of local timeline events.
+	LocalTimeline history.Timeline
```

Comment:
I'm curious why agent needs to keep 2 timelines.

Reply:
Two timelines are needed since only the master nodes are expected to store the overall cluster timeline.
The local timeline is used internally, and is only used to cache local events before pushing to master nodes.
Only the master nodes need to keep an additional cluster timeline, so we can add some conditionals to remove the cluster timeline from non master nodes.